src/Nethermind/Nethermind.Trie/PatriciaTree.cs

```diff
@@ -363,14 +347,14 @@ public void Set(Span<byte> rawKey, Rlp? value)
             {
                 throw new InvalidOperationException("Only reads can be done in parallel on the Patricia tree");
             }
-            
-#if DEBUG
-            if (nibblesCount != updatePath.Length)
-            {
-                throw new Exception("Does it ever happen?");
-            }
-#endif
-            
+
+            int nibblesCount = 2 * rawKey.Length;
+            byte[] array = null;
+            Span<byte> updatePath = rawKey.Length <= 64
+                ? stackalloc byte[nibblesCount]
+                : array = ArrayPool<byte>.Shared.Rent(nibblesCount);
```

Comment:
I would put everything between `Rent` and `Return` under `try`, and `Return` in `finally`, just to be safe.

Reply:
Will do.